platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapGestureDetector.java

```diff
@@ -449,13 +457,54 @@ public boolean onScroll(MotionEvent e1, MotionEvent e2, float distanceX, float d
       // Scroll the map
       transform.moveBy(-distanceX, -distanceY, 0 /*no duration*/);
 
-      if (onScrollListener != null) {
-        onScrollListener.onScroll();
-      }
+      notifyOnScrollListeners();
       return true;
     }
   }
 
+  void notifyOnMapClickListeners(PointF tapPoint) {
+    // notify app of map click
+    if (onMapClickListener != null) {
+      onMapClickListener.onMapClick(projection.fromScreenLocation(tapPoint));
+    }
+
+    for (MapboxMap.OnMapClickListener listener : onMapClickListenerList) {
+      listener.onMapClick(projection.fromScreenLocation(tapPoint));
+    }
+  }
+
+  void notifyOnMapLongClickListeners(PointF longClickPoint) {
+    if (!quickZoom) {
```

Comment:
I think having the quickZoom flag in the detector is a bit cleaner and conforms more to the setup of the other methods.